packages/api/src/controllers/stream.ts

```diff
@@ -193,13 +193,38 @@ export function getPlaybackUrl(ingest: string, stream: DBStream) {
   return pathJoin(ingest, `hls`, stream.playbackId, `index.m3u8`);
 }
 
-function getRecordingUrl(ingest: string, session: DBSession, mp4 = false) {
-  return pathJoin(
-    ingest,
-    `recordings`,
-    session.lastSessionId ?? session.id,
-    mp4 ? `source.mp4` : `index.m3u8`
-  );
+async function getRecordingUrl(
+  ingest: string,
+  session: DBSession,
+  mp4 = false
+) {
+  if (!session.recordingSessionId) {
+    // Backwards-compatibility for Recording V1
+    return pathJoin(
+      ingest,
+      `recordings`,
+      session.lastSessionId ?? session.id,
+      mp4 ? `source.mp4` : `index.m3u8`
+    );
+  }
+
+  // Recording V2
+  const asset = await db.asset.getBySessionId(session.id);
+  if (!asset) {
+    // Recording asset not created yet
+    return;
+  }
+  const os = await db.objectStore.get(asset.objectStoreId);
+
+  // For transcoded recordings, we store only one MP4 output file
+  const type = mp4 ? "static_transcoded_mp4" : "catalyst_hls_manifest";
+  const path = asset.files?.find((f) => f.type === type)?.path;
+  if (!path) {
+    // Recording renditions not yet ready
+    return;
+  }
+
+  return pathJoin(os.publicUrl, asset.playbackId, path);
```

Comment:
You should use [this](https://github.com/livepeer/livepeer-com/blob/ad8d909a045b536ffa7cc15955e828e6e4e6aa5d/packages/api/src/controllers/asset.ts#L227) function instead which is already exported from the asset controller. To avoid fetching the object store, you can even jut use `withPlaybackUrls` (same file, also exported) which will do that for you. It's also better cause it already checks the `asset.status.phase` field as well. Then your logic here can be `mp4 ? asset.playbackUrl : asset.downloadUrl` 😌 

You might want to rename the function on `import` so it's not just a generic "get/with playback URL" in this file which does not work for streams. [Example](https://github.com/livepeer/livepeer-com/blob/486b847193ebabc92caadc4b10db63edae533e67/packages/api/src/controllers/playback.ts#L8)

Reply:
Updated in https://github.com/livepeer/studio/pull/1692/commits/44d4cf4509b4b3c7a0c4cd326b74e7ba11aecaf2

Note that I had to change how `getDownloadUrl()` works, but I think it's a better behavior. It first tried to fetch mp4 rendition file and only if it does not exist, then it returns the source file. @victorges Let me know if it's ok in your opinion.